src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -253,39 +266,39 @@ static term setup_gpio_pin(Context *ctx, term gpio_pin_tuple, term mode_term)
 {
     if (UNLIKELY(!term_is_tuple(gpio_pin_tuple))) {
         AVM_LOGE(TAG, "Invalid GPIO Pin tuple, expect {Bank, Pin}.");
-        return create_pair(ctx, ERROR_ATOM, BADARG_ATOM);
+        return reason_atom_to_error_tuple_safely(ctx, BADARG_ATOM);
     }
     term gpio_bank_atom = term_get_tuple_element(gpio_pin_tuple, 0);
     if (UNLIKELY(!term_is_atom(gpio_bank_atom))) {
         AVM_LOGE(TAG, "Bank parameter of pin tuple must be an atom! (a...h|k depending on board)");
-        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_bank_atom));
+        return reason_atom_to_error_tuple_safely(ctx, globalcontext_make_atom(ctx->global, invalid_bank_atom));
```

Comment:
factor out idea: `error_tuple_str_maybe_gc(ctx, ATOM_STR('\x3', "foo")`

Reply:
I like the idea, but it doesn't quite work, because in some cases I am using macro defined default atoms, and in others the atoms are created as needed by the driver. Otherwise I would just take atom strings and create the atoms internally. Accepting both seems like it would overly complicate the situation.

Reply:
So, after looking at your comments about line 234 some more, and considering this recommendation, which I am really in favor of, I came up with a solution that works with both default (macro defined) atoms and atom strings.